src/boot/efi/pe.h

```diff
@@ -2,18 +2,33 @@
 #pragma once
 
 #include "efi.h"
+#include "uki.h"
 
-EFI_STATUS pe_memory_locate_sections(
-                const void *base,
-                const char * const sections[],
-                size_t *addrs,
-                size_t *sizes);
+EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address);
+
+typedef struct {
+        char name[9];
+        size_t offset;
```

Comment:
hmm, i'd don't think we need to save every byte here, let's just add a file offset field here, and a void* as a pointer. And let's make sure the latter is NULL if we never loaded the thing into memory